Src/FluentAssertions/Common/MemberPath.cs

```diff
@@ -66,8 +75,34 @@ private bool IsChildOf(MemberPath candidate)
                    && candidateSegments.SequenceEqual(Segments.Take(candidateSegments.Length));
         }
 
+        /// <summary>
+        /// Determines whether the current path is the same as <paramref name="path"/> when ignoring any specific indexes.
+        /// </summary>
+        public bool IsEquivalentTo(string path)
+        {
+            return path.WithoutSpecificCollectionIndices() == dottedPath.WithoutSpecificCollectionIndices();
+        }
+
+        public bool HasSameParentAs(MemberPath path)
+        {
+            return Segments.Length == path.Segments.Length
+                   && GetParentSegments().SequenceEqual(path.GetParentSegments());
+        }
+
+        private IEnumerable<string> GetParentSegments() => Segments.Take(Segments.Length - 1);
+
         private string[] Segments => segments ??= dottedPath.Split(new[] { '.', '[', ']' }, StringSplitOptions.RemoveEmptyEntries);
 
+        /// <summary>
+        /// Returns the name of the member the current path points to without its parent path.
+        /// </summary>
+        public string MemberName => Segments.Last();
+
+        /// <summary>
+        /// Gets a value indicating whether the current path contains an indexer like `[1]` instead of `[]`.
+        /// </summary>
+        public bool ContainsSpecificCollectionIndex => dottedPath.ContainsSpecificCollectionIndex();
```

Comment:
Well... I read the internal code and find it useful as a guideline of the cost of invoking a member.
But as it's not public, it's not a hard requirement, only a wish.

Reply:
Because it bothers you so much, I've changed it into a method ;-)